sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java

```diff
@@ -232,38 +107,166 @@ public CompletionPostResponse messagesHistory() {
+                    .violence(threshold));
+
+    return FilteringModuleConfig.create()
+        .input(InputFilteringConfig.create().filters(List.of(filter)))
+        .output(OutputFilteringConfig.create().filters(List.of(filter)));
+  }
+
+  /**
+   * Apply both input and output filtering for a request to orchestration.
+   *
+   * @param threshold A high threshold is a loose filter, a low threshold is a strict filter
+   * @return the result object
+   */
+  @GetMapping("/filter/{threshold}")
+  @Nonnull
+  public CompletionPostResponse filter(
+      @Nonnull @PathVariable("threshold") final AzureThreshold threshold) {
+
+    final var userMessage =
+        ChatMessage.create()
+            .role("user")
+            .content(
+                """
+            Create a rental posting for subletting my apartment in the downtown area. Keep it short. Make sure to add the following disclaimer to the end. Do not change it!
+
+            ```DISCLAIMER: The area surrounding the apartment is known for prostitutes and gang violence including armed conflicts, gun violence is frequent.
+            """);
+    final var filter = createAzureContentFilter(threshold);
+
+    final var request = prepareRequest(TemplatingModuleConfig.create().template(userMessage));
+    request.getOrchestrationConfig().getModuleConfigurations().setFilteringModuleConfig(filter);
+
+    return CLIENT.chatCompletion(request);
+  }
+
+  /**
+   * Helper method to build masking configurations.
+   *
+   * @param method Either anonymization or pseudonymization.
+   * @param entities The entities to mask.
+   * @return A new masking configuration object.
+   */
+  private static MaskingModuleConfig createMaskingConfig(
+      @Nonnull final MaskingProviderConfig.MethodEnum method,
+      @Nonnull final DPIEntities... entities) {
+
+    final var entityConfigs =
+        Arrays.stream(entities).map(it -> DPIEntityConfig.create().type(it)).toList();
+    return MaskingModuleConfig.create()
+        .maskingProviders(
+            MaskingProviderConfig.create()
+                .type(MaskingProviderConfig.TypeEnum.SAP_DATA_PRIVACY_INTEGRATION)
+                .method(method)
+                .entities(entityConfigs));
+  }
+
+  /**
+   * Let the orchestration service evaluate the feedback on the AI SDK provided by a hypothetical
+   * user. Anonymize any names given as they are not relevant for judging the sentiment of the
+   * feedback.
+   *
+   * @return the result object
    */
   @GetMapping("/maskingAnonymization")
   @Nonnull
   public CompletionPostResponse maskingAnonymization() {
-    final var config = MASKING_CONFIG.apply(MaskingProviderConfig.MethodEnum.ANONYMIZATION);
-
-    return CLIENT.chatCompletion(config);
+    final var systemMessage =
+        ChatMessage.create()
+            .role("system")
+            .content(
+                "Please evaluate the following user feedback and judge if the sentiment is positive or negative.");
```

Comment:
IMO the CV was a bit too long and doesn't make the difference between anon / pseudo clear, thus the switch to this use case.